src/DefaultBuilder/src/ConfigurationProviderSource.cs

```diff
@@ -64,6 +65,11 @@ public bool TryGet(string key, out string value)
                 return _provider.TryGet(key, out value);
             }
 
+            // Provide access to the original IConfigurationProvider via a single-element IEnumerable to code that goes out of its way to look for it.
+            public IEnumerator<IConfigurationProvider> GetEnumerator() => new List<IConfigurationProvider> { _provider }.GetEnumerator();
```

Comment:
I was gonna suggest you could use the compiler generated enumerable:

```C#
IEnumerable<IConfigurationProvider> Enumerable()
{
    yield return _provider;
}
```

Then do GetEnumerator from that.

Reply:
Not that I would expect this to ever be called repeatedly, but I was still curious whether this improves performance. It doubles it.

```
|                            Method |     Mean |    Error |   StdDev |  Gen 0 | Allocated |
|---------------------------------- |---------:|---------:|---------:|-------:|----------:|
|           EnumerateListEnumerable | 33.46 ns | 0.517 ns | 0.432 ns | 0.0122 |     128 B |
| EnumerateIteratorMethodEnumerable | 15.75 ns | 0.255 ns | 0.239 ns | 0.0038 |      40 B |
```

<details><summary>Benchmark code</summary>

```C#
using System.Collections;
using System.Security.Cryptography;
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

BenchmarkRunner.Run(typeof(Program).Assembly);

[MemoryDiagnoser]
public class EnumerableBenchmark
{
    private ListEnumerable _listEnumerable = new();
    private IteratorMethodEnumerable _iteratorMethodEnumerable = new();

    [Benchmark]
    public object EnumerateListEnumerable() => GetLast(_listEnumerable);

    [Benchmark]
    public object EnumerateIteratorMethodEnumerable() => GetLast(_iteratorMethodEnumerable);

    private static object GetLast(IEnumerable<object> collection)
    {
        object? result = null;

        foreach (var item in collection)
        {
            result = item;
        }

        return result;
    }

    private class ListEnumerable : IEnumerable<object>
    {
        public IEnumerator<object> GetEnumerator() => new List<object> { this }.GetEnumerator();
        IEnumerator IEnumerable.GetEnumerator() => new List<object> { this }.GetEnumerator();
    }

    private class IteratorMethodEnumerable : IEnumerable<object>
    {
        public IEnumerator<object> GetEnumerator() => IteratorMethod().GetEnumerator();
        IEnumerator IEnumerable.GetEnumerator() => IteratorMethod().GetEnumerator();

        private IEnumerable<object> IteratorMethod()
        {
            yield return this;
        }
    }
}
```
</details>